Python/bltinmodule.c

```diff
@@ -46,12 +47,85 @@ _Py_IDENTIFIER(stderr);
 
 #include "clinic/bltinmodule.c.h"
 
+static PyObject*
+update_bases(PyObject* bases, PyObject** args, int nargs)
+{
+    int i, j;
+    PyObject *base, *meth, *new_base, *result, *new_bases = NULL;
+    PyObject *stack[1] = {bases};
+    assert(PyTuple_Check(bases));
+
+    /* We have a separate cycle to calculate replacements with the idea that in
+       most cases we just scroll quickly though it and return original bases */
+    for (i = 0; i < nargs; i++) {
+        base  = args[i];
+        if (PyType_Check(base)) {
+            if (new_bases) {
+                if (PyList_Append(new_bases, base) < 0) {
+                    goto error;
+                }
+            }
+            continue;
+        }
+        if (!(meth = _PyObject_GetAttrId(base, &PyId___mro_entries__))) {
+            if (!PyErr_ExceptionMatches(PyExc_AttributeError)) {
+                goto error;
+            }
+            PyErr_Clear();
+            if (new_bases) {
+                if (PyList_Append(new_bases, base) < 0) {
+                    Py_DECREF(meth);
+                    goto error;
+                }
+            }
+            continue;
+        }
+        new_base = _PyObject_FastCall(meth, stack, 1);
+        Py_DECREF(meth);
+        if (!new_base) {
+            goto error;
+        }
+        if (PyTuple_Check(new_base)) {
+            if (!new_bases) {
+                /* If this is a first successful conversion, create new_bases list and
+                copy previously encountered bases. */
+                new_bases = PyList_New(i);
+                for (j = 0; j < i; j++) {
+                    base = args[j];
+                    PyList_SET_ITEM(new_bases, j, base);
+                    Py_INCREF(base);
+                }
+            }
+            if (!_PyList_Extend((PyListObject *)new_bases, new_base)) {
+                goto error;
+            }
+            Py_DECREF(Py_None);
+            Py_DECREF(new_base);
+        }
+        else {
+            PyErr_SetString(PyExc_TypeError,
+                            "__mro_entries__ must return a tuple");
+            Py_DECREF(new_base);
+            goto error;
+        }
+    }
+    if (!new_bases) {
+        return bases;
+    }
+    result = PyList_AsTuple(new_bases);
+    Py_DECREF(new_bases);
+    return result;
+error:
```

Comment:
This is just my personal preference, but I prefer to add an empty line after `return` and before a label (closing brace can replace an empty line). This make the code cleaner to me.